src/app/zap-templates/common/ClusterTestGeneration.js

```diff
@@ -733,14 +733,22 @@ function octetStringEscapedForCLiteral(value)
 {
   // Escape control characters, things outside the ASCII range, and single
   // quotes (because that's our string terminator).
-  return value.replace(/\p{Control}|\P{ASCII}|"/gu, ch => {
-    let code = ch.charCodeAt(0);
-    code     = code.toString(16);
+  var notASCII = [...value ].some(char => 32 > char.charCodeAt(0) || char.charCodeAt(0) > 126 || char == "\"");
```

Comment:
I would really prefer it if we didn't try to reinvent the classification wheel and just did:
```suggestion
  var needsEscape = value.match(/\p{Control}|\P{ASCII}|"/u);
```
(with `needsEscape` because `'"'` is most definitely ASCII here).

Reply:
Yep, this is much better. Thank you.